src/main/kotlin/compiler/Optimizer.kt

```diff
@@ -136,8 +142,8 @@ class Optimizer(private val coder: Coder) {
         val nulls = mutableListOf<VMWord>()
         opcodes.forEachIndexed { i, t ->
             if (t == null) nulls.add(mem[pc + i])
-            else if ((pc + i) in jumpMap.keys) hit = false  // Miss if we overlap a jump dest
-            else if (mem[pc + i].opcode != t) hit = false
+            else if (i > 0 && (pc + i) in jumpMap.keys) hit = false  // Miss if we overlap a jump dest
```

Comment:
Tiny optimization here: it's OK to optimize a group of opcodes if the first opcode is a jump dest.